src/v/cloud_storage/remote_segment.cc

```diff
@@ -225,11 +277,11 @@ ss::future<> remote_segment::run_hydrate_bg() {
     }
 }
 
-ss::future<std::filesystem::path> remote_segment::hydrate() {
+ss::future<ss::file> remote_segment::hydrate() {
```

Comment:
I was a bit confused that this return value can be different from `_data_file` - maybe just return `ss::future<>` here and use `_data_file` everywhere to indicate that we are always using one file handle?

Reply:
makes sense
fixed